zulip_bots/zulip_bots/bots/virtual_fs/virtual_fs.py

```diff
@@ -2,6 +2,20 @@
 
 import re
 import os
+import json
+
+from contextlib import ExitStack
+
+class SetEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, set):
+            return dict(_set_object=list(obj))
+        return super().default(self, obj)
+
```

Comment:
I talked to @timabbott, and I think we decided that `SetEncoder` is a bit too heavy of a solution here.  99% of bots can just use lists and dictionaries, and even this bot doesn't need the O(1)-ness of sets, since it's just a toy.  (Or even if it did, that could be done more explicitly after your read in the data.)

So I think we might need to abandon this commit.